tests/unit/test_snapshot.py

```diff
@@ -1155,6 +1168,22 @@ def test_partition_read_other_error(self):
             ),
         )
 
+    def test_partition_read_should_throw_error_for_recycled_session(self):
```

Comment:
Theoretically for paritioned reads, sessions should never be cleaned up right? It will be better to change this test. Someone reading the tests can get a different impression and think there to be bug.